builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -117,6 +122,12 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Must specify \"source\" or \"content\" field")
 	}
 
+	if _, ok := d.GetOk("kms_key_id"); ok {
+		if _, ok := d.GetOk("etag"); ok {
+			return fmt.Errorf("Unable to specify kms_key_id and etag on the same object")
+		}
+	}
+
```

Comment:
I'd try improving the message for the user, so they also understand _why_.

e.g. `Unable to specify 'kms_key_id' and 'etag' together because 'etag' wouldn't equal MD5 digest of the raw object data`